src/Dot.php

```diff
@@ -373,6 +373,33 @@ public function toJson($key = null, $options = 0)
         return json_encode($this->items, $options);
     }
 
+    /**
```

Comment:
I would tweak this a bit for efficiency and consistency on code styling. Methods are also in alphabetical order and this should follow the same pattern.

My suggestion:
```php
/**
 * Return a flatten array with a given character as key delimiter
 *
 * @param  string     $delimiter
 * @param  array|null $items
 * @param  string     $prepend
 * @return array
 */
public function flatten($delimiter = '.', $items = null, $prepend = '')
{
    $flatten = [];

    if (is_null($items)) {
        $items = $this->items;
    }

    foreach ($items as $key => $value) {
        if (is_array($value) && !empty($value)) {
            $flatten = array_merge(
                $flatten,
                $this->flatten($delimiter, $value, $prepend.$key.$delimiter)
            );
        } else {
            $flatten[$prepend.$key] = $value;
        }
    }

    return $flatten;
}
```